index.js

```diff
@@ -560,6 +560,26 @@ class ScmBase {
     _getBranchList() {
         return Promise.reject(new Error('Not implemented'));
     }
+
+    /**
+     * Open a pull request on the repository with given file change
+     *
+     * @method addWebhook
+     * @param  {Object}     config                  Configuration
+     * @param  {String}     config.scmUri           SCM URI to open pull request on
+     * @param  {String}     config.token            Service token to authenticate with the SCM service
+     * @param  {String}     config.files             Files to open pull request with
+     * @param  {String}     [config.scmContext]     The scm context name
+     * @return {Promise}                            Resolves when operation completed without failure
+     */
+    openPr(config) {
+        return validate(config, dataSchema.plugins.scm.openPr)
+            .then(() => this._openPr(config));
+    }
+
+    _openPr() {
+        return Promise.reject(new Error('Not implemented'));
```

Comment:
I don't think we should make it reject if it's not implemented
unless we plan to implement for `scm-gitlab` and `scm-bitbucket` as well

Reply:
I'd like to know the rational.

For me, I think `Promise.reject` is okay. If `scm-gitlab` and `scm-bitbucket` is not implemented, we should throw errors.

Reply:
Either ways caller should handle if pr was not opened, and UI should adjust what is displays accordingly. Not able to open PR should not be an error situation. API response should reflect that.

Reply:
Rejecting with error will mean we require all scm plugins to have this method implemented
if you do reject with error, we should bump major version of scm-base